### DISCUSSION TITLE
Adopt Cloudflares new Worker Static Assets (Beta) to run in CF Workers

### DISCUSSION THREAD

**Discussion by FraBle** [vikejs/vike#1901] [Discussion]
Cloudflare [announced](https://blog.cloudflare.com/builder-day-2024-announcements/) today better support for [full-stack (SSR) frameworks](https://developers.cloudflare.com/workers/frameworks/) on Cloudflare Workers with the new beta feature ["Worker Static Assets"](https://developers.cloudflare.com/workers/static-assets/). They already support an [extensive list of frameworks](https://developers.cloudflare.com/workers/frameworks/), but I was missing Vike in the mix 😢. Is the support for "Static Assets for Workers" on the roadmap or radar for Vike?

**Comment by brillout** [comment]
Yes, very much so. We're waiting for Vite's upcoming Environement API to be released as it will enable development to use Cloudflare Worker's runtime workerd.

CC @magne4000 who's the author of https://github.com/vikejs/vike-cloudflare.

> I was missing Vike in the mix

Yea, let's add Vike to it once we integrate Vite's Environment API.

**Comment by magne4000** [comment]
I struggle to see the differences between _Cloudflare Pages_ and this new feature for _Workers_. What can you do with _Workers_ + _Static Assets_ that you cannot with _Cloudflare Pages_ for instance?

**Reply by FraBle** [reply]
I think it comes down to what other features Workers provide: [Cloudflare Workers vs. Pages (compatibility matrix)](https://developers.cloudflare.com/workers/static-assets/compatibility-matrix/)
For example, I find Workers Logs and specifically Logpush pretty valuable.

**Reply by osseonews** [reply]
Yeah, this is a great point. Cloudflare is really making things confusing. The thing is, that originally Cloudflare was pushing Pages for frameworks and they seemed to say they would deprecate Workers for this type of functionality. Now it's the opposite, if you read the docs carefully. They are bringing more functionality to Workers and over time, you will migrate from Pages to Workers. Honestly, I wish they would just merge these two platforms to reduce the confusion. As for now, Workers does have better features over Pages. The Logs is a huge improvement as mentioned by FraBle. Personally, I don't plan to use Pages anymore and luckily I only built one website with it. Going forward, just going to build everything on Workers.

**Comment by osseonews** [comment]
Was going to post about this. Making this work on Cloudflare Workers with the new SSR framework would be amazing. Working with Vike on Cloudflare Pages is not easy for various reasons. Workers is better. Also, if you got Vike listed on the Cloudflare Worker frameowork list, it would give Vike huge exposure.
